python/lsst/ts/m2com/error_handler.py

```diff
@@ -0,0 +1,336 @@
+                    if value[1] == "Fault"
+                    else self._list_code_warning
+                )
+                list_code.append(error_code)
+
+    def clear(self) -> None:
+        """Clear all warnings and errors."""
+
+        self._errors_new.clear()
+        self._errors_reported.clear()
+
+        self._warnings_new.clear()
+        self._warnings_reported.clear()
+
+    def decode_summary_faults_status(self, status: int) -> None:
+        """Decode the summary faults status. The decoded error/warning code
+        will be put into the internal lists of error and warning.
+
+        Parameters
+        ----------
+        status : `int`
+            Summary faults status.
+        """
+        for idx, code in enumerate(self._list_code_total):
+            mask = 2**idx
+            if status < mask:
+                break
+
+            # Note some code is just dummy. Only collect the error and warning
+            # code.
+            if (status & mask) != 0:
+                if self.is_error(code):
+                    self.add_new_error(code)
+                    continue
+
+                if self.is_warning(code):
+                    self.add_new_warning(code)
+
+    def is_error(self, code: int) -> bool:
+        """Is error or not.
+
+        Parameters
+        ----------
+        code : `int`
+            Code.
+
+        Returns
+        -------
+        `bool`
+            True if the code belongs to error. Otherwise, False.
+        """
+        return code in self._list_code_error
+
+    def is_warning(self, code: int) -> bool:
+        """Is warning or not.
+
+        Parameters
+        ----------
+        code : `int`
+            Code.
+
+        Returns
+        -------
+        `bool`
+            True if the code belongs to warning. Otherwise, False.
+        """
+        return code in self._list_code_warning
+
+    def add_new_error(self, code: int) -> None:
+        """Add the new error code.
+
+        Parameters
+        ----------
+        code : `int`
+            Error code.
+        """
+        self._add_new_item(self._errors_new, self._errors_reported, code)
+
+    def _add_new_item(
+        self, set_new: set[int], set_reported: set[int], item: int
+    ) -> None:
+        """Add the new item to set.
+
+        The existed item will not be added.
+
+        Parameters
+        ----------
+        set_new : `set`
+            New set.
+        set_reported : `set`
+            Reported set.
+        item : `int`
+            Item to add.
+        """
+
+        if (item not in set_new) and (item not in set_reported):
+            set_new.add(item)
+
+    def add_new_warning(self, code: int) -> None:
+        """Add the new warning code.
+
+        Parameters
+        ----------
+        code : `int`
+            Warning code.
+        """
+        self._add_new_item(self._warnings_new, self._warnings_reported, code)
+
+    def exists_new_error(self) -> bool:
```

Comment:
Not sure I understand the meaning of these "`exists_*`" methods. Are they returning whether there are any new errors? Consider renaming them `has_*` instead, e.g.:

`has_new_errors`, `has_new_warnings`, `has_errors`, `has_warnings`...

I noticed there's a `has_warning` method, so I think this change makes sense, and the difference between `has_warnings` and `has_warning` is that the latter checks against a specific warning, whereas the former checks against all warnings.

You could also consider adding `has_error`, for completeness. 

Also, consider improving the docstrings to explain what these methods are for.

Reply:
The new `ErrorHandler` class is extracted from the original `MockErrorHandler` class. The renaming of function names will break some codes in `ts_m2gui` and `ts_m2`. In addition, I intentionally to make the function names begin from `exists_` to have the `True/False` output only without any input argument. Therefore, I would like to avoid the renaming at this moment. Thanks!